src/ConfirmationButton/ConfirmationButton.jsx

```diff
@@ -0,0 +1,59 @@
+import React, {Component, PropTypes} from "react";
+import _ from "underscore";
+
+import {Button} from "../Button/Button";
+import {Modal} from "../Modal/Modal";
+
+require("./ConfirmationButton.less");
+
+export class ConfirmationButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {showConfirmation: false};
+    this.toggleConfirmation = this.toggleConfirmation.bind(this);
+  }
+
+  toggleConfirmation(show) {
+    this.setState({showConfirmation: show});
+  }
+
+  render() {
+    const modalContents = (<div>
+      <p>{this.props.confirmationDescription}</p>
+      <p>
+        <Button type="secondary" value="Cancel"
+          onClick={_.partial(this.toggleConfirmation, false)}
```

Comment:
The partial application syntax is kind of confusing tbh, how do you feel about:
* es6 syntax: `onClick={() => this.toggleConfirmation(false)}` (this is what they do in the react docs)
* explicit handlers like `showConfirmation` and `hideConfirmation`
* just actually toggle state of taking an arg `setState({show: !this.state.show})`